testplan/web_ui/testing/src/AssertionPane/AssertionHeader.js

```diff
@@ -13,190 +13,178 @@ library.add(faLayerGroup);
-      this.props.assertion.passed === undefined
-        ? styles.cardHeaderColorLog
-        : this.props.assertion.passed
-        ? styles.cardHeaderColorPassed
-        : styles.cardHeaderColorFailed;
+  const cardHeaderColorStyle =
+    props.assertion.passed === undefined
+      ? styles.cardHeaderColorLog
+      : props.assertion.passed
+      ? styles.cardHeaderColorPassed
+      : styles.cardHeaderColorFailed;
 
-    const timeInfoArray = this.props.assertion.timeInfoArray || [];
-    let component =
-      this.props.assertion.utc_time === undefined ? (
-        <span className={css(styles.cardHeaderAlignRight)}>
-          <FontAwesomeIcon // Should be a nested assertion group
-            size="sm"
-            key="faLayerGroup"
-            icon="layer-group"
-            className={css(styles.icon)}
-          />
-        </span>
-      ) : timeInfoArray.length === 0 ? (
-        <span className={css(styles.cardHeaderAlignRight)}></span>
-      ) : (
-        <>
-          <span
-            className={css(styles.cardHeaderAlignRight, styles.timeInfo)}
-            id={`tooltip_duration_${timeInfoArray[0]}`}
-            style={{ order: 4, display: "inline-flex", alignItems: "center" }}
-          >
-            {timeInfoArray[2]}
-          </span>
-          <span
-            className={css(styles.cardHeaderAlignRight)}
-            style={{ order: 3 }}
-          >
-            &nbsp;&nbsp;
-          </span>
-          <span
-            className={css(styles.cardHeaderAlignRight, styles.timeInfo)}
-            id={`tooltip_utc_${timeInfoArray[0]}`}
-            style={{ order: 6, display: "inline-flex", alignItems: "center" }}
-          >
-            {timeInfoArray[1]}
-          </span>
-          <span
-            className={css(styles.cardHeaderAlignRight)}
-            style={{ order: 5 }}
-          >
-            &nbsp;&nbsp;
-          </span>
-          <Tooltip
-            placement="bottom"
-            isOpen={this.state.isUTCTooltipOpen}
-            target={`tooltip_utc_${timeInfoArray[0]}`}
-            toggle={this.toggleUTCTooltip}
-          >
-            {"Assertion start time"}
-          </Tooltip>
-          <Tooltip
-            placement="bottom"
-            isOpen={this.state.isDurationTooltipOpen}
-            target={`tooltip_duration_${timeInfoArray[0]}`}
-            toggle={this.toggleDurationTooltip}
-          >
-            {"Time elapsed since last assertion"}
-          </Tooltip>
-        </>
-      );
-
-    let pathButton = this.props.displayPath ? (
+  const timeInfoArray = props.assertion.timeInfoArray || [];
+  let component =
+    props.assertion.utc_time === undefined ? (
+      <span className={css(styles.cardHeaderAlignRight)}>
+        <FontAwesomeIcon // Should be a nested assertion group
+          size="sm"
+          key="faLayerGroup"
+          icon="layer-group"
+          className={css(styles.icon)}
+        />
+      </span>
+    ) : timeInfoArray.length === 0 ? (
```

Comment:
Can this be truthy/falsy in JS too?

Reply:
I believe this component needs a proper refactoring under a separate task.